### PR TITLE
Fixed bug of sometimes not showing correct events at a given restaurant

### PR DESCRIPTION
**Pull request by miajohansson9**
The on click listener for markers sometimes didn't pass in the correct Datasnapshot. To ensure that the slideupview has the correct events, I saved the restaurant's JSON object in the marker tag and used the id to get the correct events from the database. 

I also cleaned up the JSONException code.

### CODE REVIEW

File: app/src/main/java/com/example/dine_and_donate/HomeFragments/MapFragment.java [hunk 1]
```diff
@@ -448,22 +434,48 @@ public void onCancelled(DatabaseError databaseError) {
...
-                0);
-        animate.setDuration(500);
-        animate.setFillAfter(true);
-        slideView.startAnimation(animate);
-        btnCreate = slideView.findViewById(R.id.btn_create_event);
+    private boolean clickMarker(Marker marker) {
+        try {
+            final JSONObject restaurantOfMarker = (JSONObject) marker.getTag();
+            System.out.println(restaurantOfMarker);
+            // Need to make new reference on UI thread to make sure that
```
Comment (kellymckinnon) [inline app/src/main/java/com/example/dine_and_donate/HomeFragments/MapFragment.java]: make sure that what? lol

File: app/src/main/java/com/example/dine_and_donate/HomeFragments/MapFragment.java [hunk 2]
```diff
@@ -448,22 +434,48 @@ public void onCancelled(DatabaseError databaseError) {
...
-                slideView.getY(),
-                0);
-        animate.setDuration(500);
-        animate.setFillAfter(true);
-        slideView.startAnimation(animate);
-        btnCreate = slideView.findViewById(R.id.btn_create_event);
+    private boolean clickMarker(Marker marker) {
+        try {
+            final JSONObject restaurantOfMarker = (JSONObject) marker.getTag();
+            System.out.println(restaurantOfMarker);
```
Comment (kellymckinnon) [inline app/src/main/java/com/example/dine_and_donate/HomeFragments/MapFragment.java]: i'd recommend deleting printlns before pushing to master, but up to yall! @ryan-beck, @shry678

**Review by kellymckinnon — APPROVED** [review]
looks good!

File: app/src/main/java/com/example/dine_and_donate/HomeFragments/MapFragment.java (hunk 1 above)
Comment (miajohansson9) [reply app/src/main/java/com/example/dine_and_donate/HomeFragments/MapFragment.java]: oops!

**Pull request merged**